lib/manager/ansible-galaxy/extract.ts

```diff
@@ -0,0 +1,124 @@
+import { logger } from '../../logger';
+import { PackageFile, PackageDependency } from '../common';
+import * as semver from '../../versioning/semver';
+import * as git from '../../versioning/git';
+import {
+  DATASOURCE_GIT_TAGS,
+  DATASOURCE_ANSIBLE_GALAXY,
+} from '../../constants/data-binary-source';
+
+function interpretLine(
+  lineMatch: RegExpMatchArray,
+  lineNumber: number,
+  dependency: PackageDependency
+): PackageDependency {
+  const localDependency: PackageDependency = dependency;
+  const key = lineMatch[2];
+  const value = lineMatch[3].replace(/["']/g, '');
+  switch (key) {
+    case 'name': {
+      localDependency.managerData.name = value;
+      break;
+    }
+    case 'version': {
+      localDependency.managerData.version = value;
+      localDependency.currentValue = value;
+      localDependency.managerData.lineNumber = lineNumber;
+      break;
+    }
+    case 'scm': {
+      localDependency.managerData.scm = value;
+      break;
+    }
+    case 'src': {
+      localDependency.managerData.src = value;
+      break;
+    }
+    default: {
+      return null;
+    }
+  }
+  return localDependency;
+}
+
+function finalize(dependency: PackageDependency): boolean {
+  const dep = dependency;
+  if (dependency.managerData.version === null) {
+    dep.skipReason = 'no-version';
+    return false;
+  }
+
+  const source: string = dep.managerData.src;
+  const sourceMatch: RegExpMatchArray = new RegExp(
+    /^(git|http|git\+http)s?(:\/\/|@).*(\/|:)(.+\/[^.]+)\/?(\.git)?$/
+  ).exec(source);
+  if (sourceMatch) {
+    dep.datasource = DATASOURCE_GIT_TAGS;
+    dep.depName = sourceMatch[4];
+    // remove leading `git+` from URLs like `git+https://...`
+    dep.lookupName = source.replace(/git\+/, '');
+  } else if (new RegExp(/.+\..+/).exec(source)) {
+    dep.datasource = DATASOURCE_ANSIBLE_GALAXY;
+    dep.depName = dep.managerData.src;
+    dep.lookupName = dep.managerData.src;
+  } else {
+    dep.skipReason = 'no-source-match';
+    return false;
+  }
+  if (dep.managerData.name !== null) {
+    dep.depName = dep.managerData.name;
+  }
+
+  if (
+    (dep.datasource === DATASOURCE_GIT_TAGS &&
+      !git.api.isValid(dependency.managerData.version)) ||
+    (dep.datasource === DATASOURCE_ANSIBLE_GALAXY &&
+      !semver.isValid(dependency.managerData.version))
+  ) {
+    dep.skipReason = 'invalid-version';
+    return false;
+  }
+  return true;
+}
+
+export default function extractPackageFile(
```

Comment:
Because `requirements.yaml` is such a common file name, there will often be false matches. Ideally you can add some type of sanity check at the start of this routine to find files that definitely are ansible galaxy type, and return `null` early for those that aren't.

Reply:
I can not think on top of my head of a marker which we could use to identify ansible-galaxy `requriements.yml` files. 

It would be a bit hacky but we could simply check if other managers have successfully extracted `PackageDependencies` from the same file. 
An other option would be to add checks for other application which also use `requirements.yml` , but I would like to prevent this as I can hardly image that this could become a maintenance hog.

Reply:
In that case:
- Make sure that your logic and regexes are robust and don't throw errors due to assumptions about the file structure which may not be true
- Do not eagerly warn or error. Non-ansible galaxy users don't want to have their logs polluted with any "Failed to parse ansible requirements.yaml" messages if they are not even using ansible galaxy
- Make sure to return `null` if no recognisable dependencies are found

Reply:
added unit test which tests the manager against a helm requirements file